wrapper/qsl/src/main/kotlin/org/quiltmc/qsk/wrapper/qsl/client/screen/ScreenEvents.kt

```diff
@@ -0,0 +1,252 @@
+ * @see AllowMouseRelease.allowMouseRelease
+ */
+public fun EventRegistration.allowMouseRelease(callback: MouseClick<TriState>) {
+    ALLOW_MOUSE_RELEASE.register(AllowMouseRelease(callback))
+}
+
+/**
+ * @see BEFORE_MOUSE_RELEASE
+ * @see BeforeMouseRelease.beforeMouseRelease
+ */
+public fun EventRegistration.beforeMouseRelease(callback: MouseClick<Unit>) {
+    BEFORE_MOUSE_RELEASE.register(BeforeMouseRelease(callback))
+}
+
+/**
+ * @see AFTER_MOUSE_RELEASE
+ * @see AfterMouseRelease.afterMouseRelease
+ */
+public fun EventRegistration.afterMouseRelease(callback: MouseClick<Unit>) {
+    AFTER_MOUSE_RELEASE.register(AfterMouseRelease(callback))
+}
+
+/**
+ * @see ALLOW_MOUSE_SCROLL
+ * @see AllowMouseScroll.allowMouseScroll
+ */
+public fun EventRegistration.allowMouseScroll(callback: MouseScroll<TriState>) {
+    ALLOW_MOUSE_SCROLL.register(AllowMouseScroll(callback))
+}
+
+/**
+ * @see BEFORE_MOUSE_SCROLL
+ * @see BeforeMouseScroll.beforeMouseScroll
+ */
+public fun EventRegistration.beforeMouseScroll(callback: MouseScroll<Unit>) {
+    BEFORE_MOUSE_SCROLL.register(BeforeMouseScroll(callback))
+}
+
+/**
+ * @see AFTER_MOUSE_SCROLL
+ * @see AfterMouseScroll.afterMouseScroll
+ */
+public fun EventRegistration.afterMouseScroll(callback: MouseScroll<Unit>) {
+    AFTER_MOUSE_SCROLL.register(AfterMouseScroll(callback))
+}
+
+//endregion
+
+//region: Keyboard events
+
+public typealias KeyboardKey<T> = Screen.(
+    key: Int,
+    scanCode: Int,
+    modifiers: Int
+) -> T
+
+/**
+ * @see ALLOW_KEY_PRESS
+ * @see AllowKeyPress.allowKeyPress
+ */
+public fun EventRegistration.allowKeyPress(callback: KeyboardKey<TriState>) {
+    ALLOW_KEY_PRESS.register(AllowKeyPress(callback))
+}
+
+/**
+ * @see BEFORE_KEY_PRESS
+ * @see BeforeKeyPress.beforeKeyPress
+ */
+public fun EventRegistration.beforeKeyPress(callback: KeyboardKey<Unit>) {
+    BEFORE_KEY_PRESS.register(BeforeKeyPress(callback))
+}
+
+/**
+ * @see AFTER_KEY_PRESS
+ * @see AfterKeyPress.afterKeyPress
+ */
+public fun EventRegistration.afterKeyPress(callback: KeyboardKey<Unit>) {
+    AFTER_KEY_PRESS.register(AfterKeyPress(callback))
+}
+
+/**
+ * @see ALLOW_KEY_RELEASE
+ * @see AllowKeyRelease.allowKeyRelease
+ */
+public fun EventRegistration.allowKeyRelease(callback: KeyboardKey<TriState>) {
+    ALLOW_KEY_RELEASE.register(AllowKeyRelease(callback))
+}
+
+/**
+ * @see BEFORE_KEY_RELEASE
+ * @see BeforeKeyRelease.beforeKeyRelease
+ */
+public fun EventRegistration.beforeKeyRelease(callback: KeyboardKey<Unit>) {
+    BEFORE_KEY_RELEASE.register(BeforeKeyRelease(callback))
+}
+
+/**
+ * @see AFTER_KEY_RELEASE
+ * @see AfterKeyRelease.afterKeyRelease
+ */
+public fun EventRegistration.afterKeyRelease(callback: KeyboardKey<Unit>) {
+    AFTER_KEY_RELEASE.register(AfterKeyRelease(callback))
+}
+
+//endregion
+
+//region: General Screen events
+
+public typealias ScreenGenericCallback = Screen.() -> Unit
+public typealias ScreenInitCallback = Screen.(
+    mc: MinecraftClient,
```

Comment:
```suggestion
    client: MinecraftClient,
```
I'd name this `client` to match other areas of the code and this PR